src/components/sidebar/SafeList/index.tsx

```diff
@@ -180,6 +195,7 @@ const SafeList = ({ closeDrawer }: { closeDrawer?: () => void }): ReactElement =
                       owners={safe.owners.length}
                       chainId={safe.chainId}
                       closeDrawer={closeDrawer}
+                      destination={{ pathname: router.pathname, query: router.query }}
```

Comment:
Is it intended that this destination isn't doing the welcome vs non-welcome pages?

Reply:
Yes, we render safes that are neither owned nor added separately. They are only visible in the sidebar when already open so clicking on it should essentially do nothing but close the sidebar.